tensorboard/webapp/widgets/data_table/data_table_component.ng.html

```diff
@@ -20,8 +20,16 @@
           <!-- This header is intentionally left blank for the color column -->
         </th>
         <ng-container *ngFor="let header of headers;">
-          <th (click)="headerClicked(header)">
-            <div [ngSwitch]="header" class="cell">
+          <th>
+            <div
+              [draggable]="columnCustomizationEnabled"
+              (dragstart)="dragStart(header)"
+              (dragend)="dragEnd()"
+              (dragenter)="dragEnter(header)"
+              [ngSwitch]="header"
+              class="cell"
+              [ngClass]="{'highlight': header === highlightedColumn, 'highlight-border-right': header === highlightedColumn && highlightSide === Side.RIGHT, 'highlight-border-left': header === highlightedColumn && highlightSide === Side.LEFT}"
+            >
```

Comment:
Given how different this element is when `columnCustomizationEnabled` is `true`, how would you feel about rendering different templates and projecting the content?

https://angular.io/guide/content-projection

Reply:
Discussed offline and determined this is not a viable option.

However, the click event handler was added back to the header so the functionality does not change.

Reply:
I also moved the ngClass logic to a utility function.